### PR TITLE
LLM: reorganize GPU finetuning examples

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Related to https://github.com/analytics-zoo/nano/issues/973, only reorganize GPU finetuning examples in this PR, will reorganize CPU finetuning later.

### 3. Summary of the change 

- Move `templates`, `utils` and some common functions to `common` folder
- Separate different training mode to different folder (`LoRA`, `QLoRA`, `QA-LoRA`, `ReLora`)
- For each training mode folder: 
      - change the name of python script (use `alpaca_relora_finetuning.py`, etc)
      - modify the default `training_mode` value and import path of `Prompter` in each python script
      - remove unnecessary code of addressing different training mode 
      - add `export_merged_model.py`, update bash scripts and readme
- Remove old links of `example/GPU/QLoRA-FineTuning`
- Add llama2-13b/70b qlora finetuning related example scripts

### 4. How to test?
- [ ] Unit test
- [x] Local test (on 1/2 Arc, 1 tile/card of PVC)

### CODE REVIEW

File: python/llm/example/GPU/LLM-Finetuning/LoRA/lora_finetuning.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+                optimize_model=False,
+                torch_dtype=torch.bfloat16,
+                modules_to_not_convert=["lm_head"],
+            )
+        else:
+            # use bnb_config for qlora/qalora/relora, which use 4bit for base model
+            if training_mode == "qalora":
+                low_bit_format = "int4"
+            else:
+                low_bit_format = "nf4"
```
Comment (jason-dai) [inline python/llm/example/GPU/LLM-Finetuning/LoRA/lora_finetuning.py:217]: We should simply this and only support LoRA here, and refactor the common code across QLoRA, etc., to `common`

Comment (plusbang) [reply python/llm/example/GPU/LLM-Finetuning/LoRA/lora_finetuning.py:217]: > We should simply this and only support LoRA here, and refactor the common code across QLoRA, etc., to `common`

Got it, I'll extract common code to `common`.

File: python/llm/example/GPU/LLM-Finetuning/common/utils/util.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+def get_int_from_env(env_keys, default):
+    """Returns the first positive env value found in the `env_keys` list or the default."""
+    for e in env_keys:
+        val = int(os.environ.get(e, -1))
+        if val >= 0:
+            return val
+    return int(default)
+
+
+def get_trainer_cls(training_mode):
```
Comment (jason-dai) [inline python/llm/example/GPU/LLM-Finetuning/common/utils/util.py:63]: I don't think we need this any more

Comment (plusbang) [reply python/llm/example/GPU/LLM-Finetuning/common/utils/util.py]: > I don't think we need this any more

Have removed it.

**Comment by jason-dai** [comment]
Please update links in https://github.com/intel-analytics/BigDL/blob/main/README.md; otherwise looks good to me.

**Comment by plusbang** [comment]
> Please update links in https://github.com/intel-analytics/BigDL/blob/main/README.md; otherwise looks good to me.

Have updated links in `latest update` section (https://github.com/intel-analytics/BigDL/blob/7eb3cfed52d47eaa626fed47fa6338221e00af6f/README.md#latest-update-)

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**